server/providers/combined/tools/check-story-changes/core-logic.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Core business logic for check-story-changes tool
+ * 
+ * This module contains the pure business logic for analyzing divergences between
+ * a child story and its parent epic. It is independent of MCP-specific concerns
+ * (authentication, context, etc.) and can be used from both MCP handlers and REST API endpoints.
+ */
+
+import type { ToolDependencies } from '../types.js';
+import { getJiraIssue, resolveCloudId, addIssueComment } from '../../../atlassian/atlassian-helpers.js';
+import { convertAdfToMarkdown } from '../../../atlassian/markdown-converter.js';
+import type { JiraIssue } from '../../../atlassian/types.js';
+import { CHECK_STORY_CHANGES_SYSTEM_PROMPT, generateCheckWhatChangedPrompt } from './strategies/prompt-check-story-changes.js';
+import { CHECK_STORY_CHANGES_MAX_TOKENS } from './strategies/prompt-check-story-changes.js';
+
+/**
+ * Parameters for executing the check-story-changes workflow
+ */
+export interface ExecuteCheckStoryChangesParams {
+  storyKey: string;
+  cloudId?: string;
+  siteName?: string;
+}
+
+/**
+ * Result from executing the check-story-changes workflow
+ */
+export interface ExecuteCheckStoryChangesResult {
+  success: true;
+  analysis: string;
+  metadata: {
+    parentKey: string;
+    childKey: string;
+    tokensUsed?: number;
+  };
+}
+
+/**
+ * Convert Jira description to markdown text
+ * 
+ * Handles multiple description formats returned by the Jira API:
+ * - ADF (Atlassian Document Format) objects - converted to markdown
+ * - Plain strings - returned as-is
+ * - Null/undefined - returns empty string
+ * - Other objects - JSON stringified as fallback
+ * 
+ * @param description - Jira issue description (ADF object, string, or null)
+ * @returns Markdown or plain text representation of the description
+ * 
+ * @example
+ * // ADF object
+ * const adf = { type: 'doc', content: [...] };
+ * convertDescriptionToText(adf); // Returns: "# Title\n\nContent..."
+ * 
+ * @example
+ * // Plain string
+ * convertDescriptionToText("Simple description"); // Returns: "Simple description"
+ * 
+ * @example
+ * // Null or undefined
+ * convertDescriptionToText(null); // Returns: ""
+ */
+function convertDescriptionToText(description: any): string {
+  if (!description) return '';
+
+  if (typeof description === 'string') {
```

Comment:
Will it ever be a string?  

If yes, we probably should move this helper somewhere centralized. 

If not, we should just use `convertAdfToMarkdown` directly.